module1/homework/grayscaleImages/compression.cpp

```diff
@@ -0,0 +1,56 @@
+#include "compression.hpp"
+
+#include <iostream>
+
+
+compressedBitmap compressGrayscale(const image& img) {
+    compressedBitmap cb{};
+
+    for (const auto& row : img) {
+        cb.emplace_back(std::make_pair(0, 0));
+        for (const auto col : row) {
+            if (cb.back().first != col) {
+                cb.emplace_back(std::make_pair(col, 1));
+            } else {
+                ++cb.back().second;
+            }
```

Comment:
Nice, but not ideal. Try to figure out how to avoid emplace_back(0, 0) at the beginning of every row.